tox.ini

```diff
@@ -121,6 +121,24 @@ pip_pre =
     predeps: true
     !predeps: false
 
+# Test that we can build astropy with the latest developer versions of all
+# build-time dependencies,
+[testenv:devbuild]
+skip_install = true
+changedir = .tmp/{envname}
+setenv =
+    PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+deps =
+    numpy>=0.0.dev0
```

Comment:
Don't we need to unpin `numpy<2` like https://github.com/astropy/astropy/pull/15495 ?

Reply:
No because the ``--no-build-isolation`` flag means that ``pyproject.toml`` is ignored for build dependencies.